File: smartgrid/services/zone_service.py
from sqlalchemy.orm import Session
from smartgrid.repositories.zone_repository import ZoneRepository


class ZoneService:

    def __init__(self):
        self.repo = ZoneRepository()

    def get_all_zones(self, db: Session):
        return self.repo.get_all(db)

    def get_zone(self, db: Session, zone_id: int):
        return self.repo.get_by_id(db, zone_id)
    
    
    def create_zone(self, db: Session, zone):
        return self.repo.create(db, zone)

    def update_zone(self, db: Session, zone_id: int, zone):
        return self.repo.update(db, zone_id, zone)

    def delete_zone(self, db: Session, zone_id: int):
        return self.repo.delete(db, zone_id)

File: frontend/src/services/zoneService.js
import api from "./api";

// Get all zones
export const getZones = async () => {
    const response = await api.get("/zones");
    return response.data;
};

// Get single zone
export const getZone = async (id) => {
    const response = await api.get(`/zones/${id}`);
    return response.data;
};

// Create zone
export const createZone = async (zoneData) => {
    const response = await api.post("/zones", zoneData);
    return response.data;
};

// Update zone
export const updateZone = async (id, zoneData) => {
    const response = await api.put(`/zones/${id}`, zoneData);
    return response.data;
};

// Delete zone
export const deleteZone = async (id) => {
    const response = await api.delete(`/zones/${id}`);
    return response.data;
};